crates/polars-lazy/src/tests/io.rs

```diff
@@ -440,37 +440,66 @@ fn slice_at_union(lp_arena: &Arena<ALogicalPlan>, lp: Node) -> bool {
 }
 
 #[test]
-fn test_csv_globbing() -> PolarsResult<()> {
+fn test_csv_globbing() {
     let glob = "../../examples/datasets/foods*.csv";
-    let full_df = LazyCsvReader::new(glob).finish()?.collect()?;
+    let full_df = LazyCsvReader::new(glob)
+        .finish()
+        .unwrap()
+        .collect()
+        .unwrap();
 
     // all 5 files * 27 rows
     assert_eq!(full_df.shape(), (135, 4));
-    let cal = full_df.column("calories")?;
-    assert_eq!(cal.get(0)?, AnyValue::Int64(45));
-    assert_eq!(cal.get(53)?, AnyValue::Int64(194));
-
-    let glob = "../../examples/datasets/*.csv";
-    let lf = LazyCsvReader::new(glob).finish()?.slice(0, 100);
+    let cal = full_df.column("calories").unwrap();
+    assert_eq!(cal.get(0).unwrap(), AnyValue::Int64(45));
+    assert_eq!(cal.get(53).unwrap(), AnyValue::Int64(194));
+}
 
-    let df = lf.clone().collect()?;
+#[test]
+fn test_csv_globbing_2() {
+    let lf = LazyCsvReader::new("../../examples/datasets/*.csv")
+        .finish()
+        .unwrap()
+        .slice(0, 100);
+    let df = lf.clone().collect().unwrap();
     assert_eq!(df.shape(), (100, 4));
-    let df = LazyCsvReader::new(glob).finish()?.slice(20, 60).collect()?;
-    assert!(full_df.slice(20, 60).equals(&df));
+}
```

Comment:
I expect this one to fail actually because one of the csv's ([`null_nutriscore.csv`](https://github.com/pola-rs/polars/blob/96600d37d1a6ebd80210d1f057b3bbb41162b7bf/examples/datasets/null_nutriscore.csv#L14)) has 6 columns whereas the `foods*.csv` have 4.

Reply:
This test should only glob the `food*.csv`. 

The `null_nutriscore.csv` was added later. We should ensure the implementation fails if a schema is different.

Reply:
I changed this test to only use `food*.csv`, and added another test that ensures that globbing with different schemas (if all files are hit) results in an error.